pandas/core/generic.py

```diff
@@ -7872,6 +7872,14 @@ def isna(self: NDFrameT) -> NDFrameT:
         Everything else gets mapped to False values. Characters such as empty
         strings ``''`` or :attr:`numpy.inf` are not considered NA values
         (unless you set ``pandas.options.mode.use_inf_as_na = True``).
+        Note: For :class:`FloatingArray` and float-type
+        :class:`ArrowExtensionArray` the behavior is different, as they
+        use a mask (represented as pd.NA values) to designate missing
+        data but can also hold NaN which will **not** get mapped to True
+        (these values can be detected by :func:`numpy.isnan`).
+        Users may find this behavior unintuitive and want to
+        avoid these dtypes if they prefer to have only one
+        type of missing/NA value.
```

Comment:
Personally, I would leave this part out as this is quite subjective (and also not necessarily correct, depending on your view point: FloatingDtype _has_ only one type of missing data, i.e. NA, and NaN is not considered as a missing value. And I of course know that most pandas users will regard NaN as a missing value, since that's basically the main way we implemented missing values for a long time).

Reply:
It’s true that the advice is subjective and not merely describing the behavior. I thought it was useful to give practical guidance to the common type of user (with common existing code), and it was the response given to me on the discussion issue. But maybe this isn’t the right place to include - happy to drop it.